lib/lte_link_control/lte_lc.c

```diff
@@ -886,6 +886,39 @@ int lte_lc_edrx_req(bool enable)
 	return err;
 }
 
+int lte_lc_pdp_context_set(enum lte_pdp_type type, const char *apn,
+			bool ip4_addr_alloc, bool nslpi, bool securePCO)
+{
+	static const char * const pdp_type_lut[] = {
+		"IP", "IPV6", "IPV4V6"
+	};
+
+	if (apn == NULL || type > PDP_TYPE_IPV4V6) {
+		return -EINVAL;
+	}
+
+	snprintf(cgdcont, sizeof(cgdcont),
+			"AT+CGDCONT=0,\"%s\",\"%s\",0,0,0,%d,0,0,0,%d,%d,%d",
+			pdp_type_lut[type], apn,
+			ip4_addr_alloc, nslpi, securePCO);
+
+	return 0;
+}
+
+int lte_lc_pdn_auth_set(u8_t auth_prot,
+			const char *username, const char *password)
+{
+	if (username == NULL || password == null || auth_prot > 3) {
```

Comment:
Why `3`? Please avoid magic numbers.

Reply:
It is based on nRF91 AT Commands doc.

Reply:
Should actually be `> 2`

Reply:
I see, AT Commands reference is not very descriptive either. I think it would be good to create an enum with allowed values, and then verify if the argument does not exceed the largest, would be way more readable.
```
enum pdn_auth_type {
	PDN_AUTH_TYPE_NONE = 0,
	PDN_AUTH_TYPE_PAP = 1,
	PDN_AUTH_TYPE_CHAP = 2,
}
```
The function should accept `enum pdn_auth_type` as an argument type then, and we could compare with `PDN_AUTH_TYPE_CHAP`.

Reply:
Agreed on the use of enums to make it less error prone and more user-friendly